lib/rules/local-dependency.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @fileoverview Checks existence of local dependencies in the package.json
+ * @author Kendall Gassner
+ */
+'use strict';
+const fs = require('fs');
+const path = require('path');
+const {
+    isPackageJson,
+    extractPackageObjectFromAST
+} = require('../processors/PackageJsonProcessor');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+function fileExistsWithCaseSync(filepath) {
```

Comment:
[Suggestion] Resolving a path to a package can be pretty tricky 😄. I would think `require.resolve` would work instead of the body of this function, to avoid a recursive call. Have you tried that?

Note that ESM environments might need https://nodejs.org/api/module.html#modulecreaterequirefilename...